File: engine/model/src/main/scala/net/server/channel/packet/ring/EngagementProposalPacket.scala
package net.server.channel.packet.ring

class EngagementProposalPacket(private var _mode: Byte, private var _name: String, private var _itemId: Int) extends BaseRingPacket(_mode = _mode) {
  def name: String = _name

  def itemId: Int = _itemId
}

File: engine/engine-model/src/main/java/net/server/channel/packet/ring/EngagementProposalPacket.java
package net.server.channel.packet.ring;

public class EngagementProposalPacket extends BaseRingPacket {
   private final String name;

   private final Integer itemId;

   public EngagementProposalPacket(Byte mode, String name, Integer itemId) {
      super(mode);
      this.name = name;
      this.itemId = itemId;
   }

   public String name() {
      return name;
   }

   public Integer itemId() {
      return itemId;
   }
}
